### ISSUE TITLE
Change domain 

### ISSUE THREAD

**Issue by Happyfeet01** [tootsuite/mastodon#14984]
Hello, is it possible to change the domain of an instance? How it works?
I changed the URL at the server settings. Also in the env.production file and nginx. But I got an redirect error.

**Comment by ThibG** [comment]
No, it is not possible, because other Mastodon instances (and other software as well) expect the identifiers not to change. Changing either `LOCAL_DOMAIN` or `WEB_DOMAIN` for an already-running instance will cause many discrepancies across instances and may even prevent the existing accounts from working at all for instances that used to know them.

**Comment by mashirozx** [comment]
how about run the database backup on a mirror site with the new domain, and creat alias and migrate for every account with a script or tootctl? :D

**Comment by ThibG** [comment]
It wouldn't completely work, as the identifier of every activity (mainly toots) and users would change. This means your instance would think it has boosted something remote while that remote instance will only see the “old you” having boosted it, so you can't unboost it etc.

A bit more of an issue still is regarding follows: your instance would think your remote followers are still following you, while the remote followers would not.

That being said, if you accept the inconsistency regarding toots and accept dropping all your remote followers (through if done correctly, with the migration mechanism, most of them should end up following your “new” accounts), this could work with a bit of care.

**Comment by Happyfeet01** [comment]
I use this tutorial. Can I run 2 instances on one server. And then migrate from one instance to the other?
https://www.howtoforge.com/how-to-install-mastodon-social-network-with-docker-on-ubuntu-1804/#step-install-mastodon

Must I change some ports?

**Comment by mariobreskic** [comment]
> I use this tutorial. Can I run 2 instances on one server. And then migrate from one instance to the other? https://www.howtoforge.com/how-to-install-mastodon-social-network-with-docker-on-ubuntu-1804/#step-install-mastodon
> 
> Must I change some ports?

I am building this in my head right now as well, because I want to change the domain name my mastodon can be reached at. I have followed the same tutorial with a few changes to set up my current mastodon server, so I hope that creating a second mastodon server and then exporting my data from the old one to the new one would work without too much trouble.

I would just change the port to old_port++ for the new instance, otherwise both instances would occupy the same port?

Did you manage to migrate successfully in the interim?
